eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -79,24 +116,42 @@ private SubmittedVariantEntity getSubmittedVariantEntity(Document notClusteredSu
 
     @Override
     public void open(ExecutionContext executionContext) throws ItemStreamException {
+        if (executionContext.containsKey(CURRENT_ID_KEY)) {
```

Comment:
An idea for the future might be to check if this is persisted across multiple runs of a job. If so, we can leverage it to resume jobs from where we left off.

Reply:
I was wondering this as well, for jobs I ran in dev it's persisted as part of the step's execution context but not the job's execution context... I'm not sure if that's enough for Spring to use it to restart or whether we need to propagate it to the job level, but it would be cool to get this working!

I don't think it would add any overhead since Spring is already doing some context persistence, we're just adding a couple fields.  Others might know more about this though...